velox/exec/HashBuild.cpp

```diff
@@ -912,11 +914,19 @@ void HashBuild::setupSpillInput(HashJoinBridge::SpillInput spillInput) {
 void HashBuild::processSpillInput() {
   checkRunning();
 
+  // TODO: integrate with driver yield framework.
```

Comment:
Are you thinking of doing something similar to TableScan?

```
if (this->driverCtx_->task->shouldStop() != StopReason::kNone or
          (getOutputTimeLimitMs_ != 0 and
           (getCurrentTimeMs() - startTimeMs) >= getOutputTimeLimitMs_)) {
```

Reply:
I think the driver could have save the timestamp that a driver started to run and provide a advisory method for operator to check in some busy loop to see if they need to jump out. We might want to override the default in some special case. WDYT? Thanks!

Reply:
This would be nice. Thanks.